server/src-lib/Hasura/Server/Query.hs

```diff
@@ -99,25 +100,28 @@ $(deriveJSON
   ''RQLQuery)
 
 newtype Run a
-  = Run {unRun :: StateT SchemaCache (ReaderT (UserInfo, HTTP.Manager, SQLGenCtx) (LazyTx QErr)) a}
+  = Run {unRun :: StateT SchemaCache (ReaderT (UserInfo, HTTP.Manager, SQLGenCtx, FeatureFlags) (LazyTx QErr)) a}
   deriving ( Functor, Applicative, Monad
            , MonadError QErr
            , MonadState SchemaCache
-           , MonadReader (UserInfo, HTTP.Manager, SQLGenCtx)
+           , MonadReader (UserInfo, HTTP.Manager, SQLGenCtx, FeatureFlags)
            , CacheRM
            , CacheRWM
            , MonadTx
            , MonadIO
            )
 
 instance UserInfoM Run where
-  askUserInfo = asks _1
+  askUserInfo = asks (\(userInfo, _, _, _) -> userInfo)
 
 instance HasHttpManager Run where
-  askHttpManager = asks _2
+  askHttpManager = asks (\(_, httpManager, _, _) -> httpManager)
 
 instance HasSQLGenCtx Run where
-  askSQLGenCtx = asks _3
+  askSQLGenCtx = asks (\(_, _, sqlGenCtx, _) -> sqlGenCtx)
+
+instance HasFeatureFlags Run where
+  askFeatureFlags = asks (\(_, _, _, featureFlags) -> featureFlags)
```

Comment:
Use `(^. _4)` (from Control.Lens module) instead of `\(_, _, _, featureFlags) -> featureFlags`. Similarly for `askSQLGenCtx` etc.